build-system/tasks/compile.js

```diff
@@ -353,98 +346,99 @@ function compile(entryModuleFilenames, outputDir, outputFilename, options) {
-        externs,
-        js_module_root: [
-          // Do _not_ include 'node_modules/' in js_module_root with 'NODE'
-          // resolution or bad things will happen (#18600).
-          'build/patched-module/',
-          'build/fake-module/',
-          'build/fake-polyfills/',
-        ],
-        entry_point: entryModuleFilenames,
-        module_resolution: 'NODE',
-        process_common_js_modules: true,
-        // This strips all files from the input set that aren't explicitly
-        // required.
-        only_closure_dependencies: true,
-        output_wrapper: wrapper,
-        create_source_map: intermediateFilename + '.map',
-        source_map_location_mapping:
-            '|' + sourceMapBase,
-        warning_level: options.verboseLogging ? 'VERBOSE' : 'DEFAULT',
-        jscomp_error: [],
-        // moduleLoad: Demote "module not found" errors to ignore missing files
-        //     in type declarations in the swg.js bundle.
-        jscomp_warning: ['moduleLoad'],
-        // Turn off warning for "Unknown @define" since we use define to pass
-        // args such as FORTESTING to our runner.
-        jscomp_off: ['unknownDefines'],
-        define,
-        hide_warnings_for: hideWarningsFor,
-      },
+      compilation_level: options.compilationLevel || 'SIMPLE_OPTIMIZATIONS',
+      // Turns on more optimizations.
+      assume_function_wrapper: true,
+      // Transpile from ES6 to ES5.
+      language_in: 'ECMASCRIPT6',
+      language_out: 'ECMASCRIPT5',
+      // We do not use the polyfills provided by closure compiler.
+      // If you need a polyfill. Manually include them in the
+      // respective top level polyfills.js files.
+      rewrite_polyfills: false,
+      externs,
+      js_module_root: [
+        // Do _not_ include 'node_modules/' in js_module_root with 'NODE'
+        // resolution or bad things will happen (#18600).
+        'build/patched-module/',
+        'build/fake-module/',
+        'build/fake-polyfills/',
+      ],
+      entry_point: entryModuleFilenames,
+      module_resolution: 'NODE',
+      process_common_js_modules: true,
+      // This strips all files from the input set that aren't explicitly
+      // required.
+      only_closure_dependencies: true,
+      output_wrapper: wrapper,
+      source_map_location_mapping: '|' + sourceMapBase,
+      warning_level: options.verboseLogging ? 'VERBOSE' : 'DEFAULT',
+      jscomp_error: [],
+      // moduleLoad: Demote "module not found" errors to ignore missing files
+      //     in type declarations in the swg.js bundle.
+      jscomp_warning: ['moduleLoad'],
+      // Turn off warning for "Unknown @define" since we use define to pass
+      // args such as FORTESTING to our runner.
+      jscomp_off: ['unknownDefines'],
+      define,
+      hide_warnings_for: hideWarningsFor,
     };
 
     // For now do type check separately
-    if (argv.typecheck_only || checkTypes) {
+    if (options.checkTypes || options.typeCheckOnly) {
```

Comment:
Is `argv` option not used anywhere?

Reply:
I searched through the code, and it isn't documented or used anywhere. As of today, `gulp dist --typecheck_only` has been replaced by `gulp check-types`.